### PR TITLE
support ARM64

### PR DESCRIPTION
**Pull request by spacewander**

### CODE REVIEW

File: t/exdata.t [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+local u64 = ffi.new("uintptr_t", 0xefdeadbeefLL)
+local ptr = ffi.cast("void *", u64)
+local ptr2 = ffi.cast("void *", u64 + 1)
+local ptr3 = ffi.cast("void *", u64 - 2)
+local saved_q
+local function f()
+  coroutine.yield(exdata())
+  exdata(ptr2)
+  coroutine.yield(exdata())
+  coroutine.yield(exdata())
```
Comment (thibaultcha) [inline t/exdata.t:53]: minor style: this function uses 2 lines indentation, instead of 4 in the rest of this file - should we update it?

**Comment by agentzh** [comment]
@spacewander Will you please drop a note to README for this new `thread.exdata` Lua API?

**Comment by spacewander** [comment]
@agentzh 
Updated.

File: t/exdata.t (hunk 1 above)
Comment (agentzh) [reply t/exdata.t:53]: @thibaultcha Oh, yeah. Good catch. @spacewander Will you please adjust the indentation here a bit? Many thanks!

Comment (agentzh) [reply t/exdata.t:53]: @spacewander Never mind, I'll do it :)

**Comment by agentzh** [comment]
Merged with the following extra patch:

```diff
diff --git a/README b/README
index b7dc5ad352..159ee994a4 100644
--- a/README
+++ b/README
@@ -31,7 +31,7 @@ since we still synchronize any upstream changes all the time.
 We introduce our own changes which will never merge or haven't yet merged into
 the upstream LuaJIT (https://github.com/LuaJIT/LuaJIT), which are

-feature: implemented lua thread exdata.
+feature: implemented the new Lua and C API functions for thread exdata.

     The Lua API can be used like below:

@@ -56,6 +56,9 @@ feature: implemented lua thread exdata.
     but still have their own exdata storage. So child Lua threads can always
     override the inherited parent exdata pointer values.

+    This API is used internally by the OpenResty core so never ever mess
+    with it yourself in the context of OpenResty.
+
 * feature: luajit.h: defined the macro OPENRESTY_LUAJIT for our branch of
   LuaJIT.

diff --git a/t/exdata.t b/t/exdata.t
index 722fcbfb23..239bb86c18 100644
--- a/t/exdata.t
+++ b/t/exdata.t
@@ -47,10 +47,10 @@ local ptr2 = ffi.cast("void *", u64 + 1)
 local ptr3 = ffi.cast("void *", u64 - 2)
 local saved_q
 local function f()
-  coroutine.yield(exdata())
-  exdata(ptr2)
-  coroutine.yield(exdata())
-  coroutine.yield(exdata())
+    coroutine.yield(exdata())
+    exdata(ptr2)
+    coroutine.yield(exdata())
+    coroutine.yield(exdata())
 end

 exdata(u64)
```

Thanks!

**Pull request closed without merging**